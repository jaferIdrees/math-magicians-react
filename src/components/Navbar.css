.navBar {
  position: relative;
  margin: 20px;
  float: right;
  display: flex;
  flex-direction: row;
  list-style: none;
}

.navBar li {
  margin : 0;
  padding: 0.5rem;
  font-size: 1.5rem;
}

.navBar li:nth-child(1),
.navBar li:nth-child(2) {
  border-right: 3px solid rgb(2, 61, 5);
}

.mainTitle {
  position: relative;
  margin: 20px;
  float: left;
  padding: 0.5rem;
}