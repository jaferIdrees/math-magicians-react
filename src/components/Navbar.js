import { Link } from 'react-router-dom';
import './Navbar.css';

const links = [
  {
    id: 1,
    path: '/',
    text: 'Home',
  },
  {
    id: 2,
    path: '/calculator',
    text: 'Calculator',
  },
  {
    id: 3,
    path: '/quote',
    text: 'Quote',
  },
];

function Navbar() {
  return (
    <div className="navbarContainer">
      <nav>
        <ul className="navBar">
          {links.map((link) => (
            <li key={link.id}>
              <Link to={link.path}>{link.text}</Link>
            </li>
          ))}
        </ul>
      </nav>
      <h1 className="mainTitle">Math Magicians</h1>
    </div>
  );
}

export default Navbar;
