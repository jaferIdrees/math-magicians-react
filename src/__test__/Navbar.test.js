import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';

describe('Navbar component testing using snapshots', () => {
  it('Test Navbar component renders', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>,
      );
    expect(tree).toMatchSnapshot();
  });
});
