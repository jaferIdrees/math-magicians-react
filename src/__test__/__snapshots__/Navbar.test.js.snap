// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component testing using snapshots Test Navbar component renders 1`] = `
<div
  className="navbarContainer"
>
  <nav>
    <ul
      className="navBar"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
  <h1
    className="mainTitle"
  >
    Math Magicians
  </h1>
</div>
`;
